### PR TITLE
cmd/gocritic: add checker-new command

### PR DESCRIPTION
**Pull request by Quasilyte**

### CODE REVIEW

**Comment by Quasilyte** [comment]
I'll fix lint error later. This is not high priority and can be delayed.

File: cmd/gocritic/main.go [hunk 1]
```diff
@@ -34,6 +35,11 @@ var subCommands = []*subCommand{
 		name:  "check-project",
 		short: "run gocritic over specified source tree, recursively",
 	},
+	{
+		main:  checkertools.NewMain,
+		name:  "checker-new",
```
Comment (cristaloleg) [inline cmd/gocritic/main.go]: Not PR related, but: what about renaming commands to `package`, `project`, `newlinter` ?
Till we've some time before public release.

File: cmd/checkertools/new.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		log.Printf("usage: rulename")
+		flag.PrintDefaults()
+	}
+	flag.Parse()
+
+	if flag.NArg() != 1 {
+		flag.Usage()
+		os.Exit(1)
+	}
+	// TODO: add camel case check.
```
Comment (cristaloleg) [inline cmd/checkertools/new.go]: I think we can add note in usage about camel case style and if name doesn't fit proper style we can cover this during review before merging 😉

File: cmd/checkertools/new.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	return testDir
+}
+
+func createPositiveTests(testDir string) {
+	const data = `package checker_test
+
+// TODO: write tests that trigger newly added checker.
+
+/// this is example of warning directive. It fails your tests right now
```
Comment (cristaloleg) [inline cmd/checkertools/new.go]: What about note about `///` ? 😉

Comment (cristaloleg) [reply cmd/checkertools/new.go]: Or explain what this line should contain, ex warning message.

File: cmd/checkertools/new.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+// Optional section for details.
+//
+// Before:
+// badCode := "minimal example of code that triggers a warning"
+//
+// After:
+// goodCode := "example of intended fix"
+
+func init() {
+	addChecker(&{{.TypeName}})
```
Comment (cristaloleg) [inline cmd/checkertools/new.go]: What about comments regarding attributes? and notice to remove it 😉

Comment (cristaloleg) [reply cmd/checkertools/new.go]: However it's not important.

File: cmd/gocritic/main.go (hunk 1 above)
Comment (Quasilyte) [reply cmd/gocritic/main.go]: I'm not sure. Can't alias solve problems for users that don't like typing extra `len("check-")`?

File: cmd/checkertools/new.go (hunk 3 above)
Comment (Quasilyte) [reply cmd/checkertools/new.go]: We should decide whether this is a tutorial or stub generator.
Or add `-with-help` flag that is true by default.

**Comment by dmirogin** [comment]
Does we really want to make this command hmm..., public?
When somebody install gocritic then he will see checker-new command together with other helpfull for user commands ?

**Comment by Quasilyte** [comment]
@dmirogin, agreed.
Also, it's probably simpler to do it from scratch since there were multiple big changes since this PR.

**Pull request closed without merging**